libraries/stumbler/src/main/java/org/mozilla/mozstumbler/service/core/http/HttpUtil.java

```diff
@@ -220,13 +213,13 @@ public IResponse post(String urlString, byte[] data, Map<String, String> headers
 
 
         try {
-            httpURLConnection = (HttpURLConnection) url.openConnection();
+            httpURLConnection = (HttpsURLConnection) mozOpenConnection(url);
             httpURLConnection.setConnectTimeout(5000); // set timeout to 5 seconds
             // HttpURLConnection and Java are braindead.
             // http://stackoverflow.com/questions/8587913/what-exactly-does-urlconnection-setdooutput-affect
             httpURLConnection.setDoOutput(true);
             httpURLConnection.setRequestMethod("POST");
-        } catch (IOException e) {
+        } catch (Exception e) {
             ClientLog.e(LOG_TAG, "Couldn't open a connection: ", e);
```

Comment:
Oh wait. I'm just tired and ClinetLog will just do the right thing if there is no GUI.